src/params/progression.jl

```diff
@@ -34,17 +42,38 @@ function sample_severity(rng::AbstractRNG, age::Real)
   elseif age < 80;  dist = severity_dists[5]
   else;             dist = severity_dists[6]
   end
+  idx = gender + 1 |> Int32
+  prob = hospitalization_probs[idx][age < max_age_hosp ? age + 1 : max_age_hosp]
+  if age < 0;       error("age should be non-negative")
+  elseif age < 40;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[1]) ,  prob *critical_probs[1]])
+  elseif age < 50;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[2]) ,  prob *critical_probs[2]])
+  elseif age < 60;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[3]) ,  prob *critical_probs[3]])
+  elseif age < 70;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[4]) ,  prob *critical_probs[4]])
+  elseif age < 80;  dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[5]) ,  prob *critical_probs[5]])
+  else;             dist = Categorical([0,  1.0 - prob,  prob *(1-critical_probs[6]) ,  prob *critical_probs[6]])
+  end
   severity_int = rand(rng, dist)
   severity = rand(rng, dist) |> Severity
 end
 
-function sample_if_death(rng::AbstractRNG, severity::Severity)
+function sample_if_death(rng::AbstractRNG, severity::Severity, age::Real)
   severity_int = severity |> UInt8
   death_prob = death_probs[severity_int]
+  if severity != Critical
+    return false
+  elseif age < 0;       error("age should be non-negative")
+  elseif age < 40;  death_prob = death_probs_age[1]
+  elseif age < 50;  death_prob = death_probs_age[2]
+  elseif age < 60;  death_prob = death_probs_age[3]
+  elseif age < 70;  death_prob = death_probs_age[4]
+  elseif age < 80;  death_prob = death_probs_age[5]
+  else;             death_prob = death_probs_age[6]
```

Comment:
if you use `agegroup` function this will become simple `death_prob=death_probs_age[agegroup(thresholds, age)]` (maybe it is better to split into two lines ;) )

Reply:
[Here](https://github.com/MOCOS-COVID19/MocosSim/blob/bf6d061b325873f95b76084db46be5440dd43afc/src/params/age_coupling.jl#L8) is the `agegroup` function being used